builder/vsphere/common/step_export.go

```diff
@@ -314,8 +372,79 @@ func (s *StepExport) Run(ctx context.Context, state multistep.StateBag) multiste
 		return multistep.ActionHalt
 	}
 
-	// Completed exporting the virtual machine image to Open Virtualization Format (OVF).
-	ui.Say("Completed export to Open Virtualization Format (OVF).")
+	// Check the export format to determine if the image should be converted.
+	switch s.Format {
+	case "", "ovf":
+		ui.Say(fmt.Sprintf("Completed export to Open Virtualization Format (OVF): %s", s.Name+".ovf"))
+		return multistep.ActionContinue
+	case "ova":
+		ovftool := getOvftool()
+		ovaTarget := getTarget(s.OutputDir, s.Name, ".ova")
+
+		// If the OVA file already exists, remove it.
+		if s.Force {
+			_, err := os.Stat(ovaTarget)
+			if err == nil {
+				ui.Say(fmt.Sprintf("Force export enabled; removing existing OVA file: %s...", s.Name+".ova"))
+				err := os.Remove(ovaTarget)
+				if err != nil {
+					state.Put("error", errors.Wrap(err, "unable to remove existing ova file"))
+					return multistep.ActionHalt
+				}
+			} else if !errors.Is(err, os.ErrNotExist) {
+				state.Put("error", errors.Wrap(err, "unable to check if ova file exists"))
+				return multistep.ActionHalt
+			}
+		}
+
+		// Convert the Open Virtualization Format (OVF) to Open Virtualization Archive (OVA).
+		ui.Say("Converting to Open Virtualization Archive (OVA)...")
+		cmd := exec.Command(ovftool, target, ovaTarget)
+		err = cmd.Run()
+		if err != nil {
+			state.Put("error", errors.Wrap(err, "unable to convert ovf to ova"))
+			return multistep.ActionHalt
+		}
+
+		// Check if the OVA file exists.
+		_, err = os.Stat(ovaTarget)
+		if os.IsNotExist(err) {
+			state.Put("error", errors.New("unable to convert ovf to ova; ova file not found"))
+			return multistep.ActionHalt
+		}
+
+		// Clean up the files used for the conversion.'
+		ui.Say("Removing intermediate files...")
+
+		// Removes the .vmdk files.
+		for _, file := range cdp.OvfFiles {
+			absPath := filepath.Join(s.OutputDir, file.Path)
+			ui.Say(fmt.Sprintf("Removing %s...", file.Path))
+			err := os.Remove(absPath)
+			if err != nil {
+				if errors.Is(err, os.ErrNotExist) {
```

Comment:
Out of curiosity, since below we're iterating on constants I can comprehend why we'd want to check the error type before printing the warning/error, but in this case since it's from a structure's attribute (not sure how it's populated), I wonder if we can remove this error check in the end?
To be clear it doesn't hurt to have it, but I wonder if we have a reason to keep it.

Reply:
Good point, perhaps I was being overly cautious here, but since since it's coming back from `cdp.OvfFiles` it seems unecessary.

Reply:
Resolved in 97a2527.